core/geometry/src/test/geometry3d/Ray3d.test.ts

```diff
@@ -282,3 +288,297 @@ describe("Ray3d", () => {
+    }
+    ray.cloneTransformed(rotationTransform, rotatedRay);
+    triangle.cloneTransformed(rotationTransform, rotatedTriangle);
+    rotatedIntersectionPoint = rotatedRay.intersectionWithTriangle(
+      rotatedTriangle.points[0], rotatedTriangle.points[1], rotatedTriangle.points[2]
+    );
+    rotatedOriginalIntersectionPoint = rotationMatrix.multiplyPoint(intersectionPoint);
+    if (ck.testDefined(rotatedIntersectionPoint)) {
+      ck.testPoint3d(
+        rotatedOriginalIntersectionPoint,
+        rotatedIntersectionPoint!,
+        "rotating original intersection points gives rotated intersection points"
+      );
+    }
+    captureRay(allGeometry, ray, rotatedRay);
+    captureTriangle(allGeometry, triangle, rotatedTriangle);
+
+    origin = Point3d.create(5, 0, -2);
+    direction = Vector3d.create(0, 0, 1);
+    ray = Ray3d.create(origin, direction);
+    intersectionPoint = ray.intersectionWithTriangle(triangle.points[0], triangle.points[1], triangle.points[2])!;
+    expectedIntersectionPoint = Point3d.create(5, 0, 0);
+    if (ck.testDefined(intersectionPoint)) {
+      ck.testPoint3d(
+        intersectionPoint, expectedIntersectionPoint, "ray intersects triangle on a triangle edge"
+      );
+    }
+    ray.cloneTransformed(rotationTransform, rotatedRay);
+    triangle.cloneTransformed(rotationTransform, rotatedTriangle);
+    rotatedIntersectionPoint = rotatedRay.intersectionWithTriangle(
+      rotatedTriangle.points[0], rotatedTriangle.points[1], rotatedTriangle.points[2]
+    );
+    rotatedOriginalIntersectionPoint = rotationMatrix.multiplyPoint(intersectionPoint);
+    if (ck.testDefined(rotatedIntersectionPoint)) {
+      ck.testPoint3d(
+        rotatedOriginalIntersectionPoint,
+        rotatedIntersectionPoint!,
+        "rotating original intersection points gives rotated intersection points"
+      );
+    }
+    captureRay(allGeometry, ray, rotatedRay);
+
+    origin = Point3d.create(5, 0, -2);
+    direction = Vector3d.create(0, 0, 1);
+    ray = Ray3d.create(origin, direction);
+    triangle = BarycentricTriangle.createXYZXYZXYZ(2, 0, 0, 2, 0, 0, 2, 10, 0);
+    intersectionPoint = ray.intersectionWithTriangle(triangle.points[0], triangle.points[1], triangle.points[2]);
+    ck.testUndefined(
+      intersectionPoint,
+      "expect no intersection when we have a degenerate triangle with two equal vertexes"
+    );
+
+    origin = Point3d.create(5, 0, -2);
+    direction = Vector3d.create(0, 0, 1);
+    ray = Ray3d.create(origin, direction);
+    triangle = BarycentricTriangle.createXYZXYZXYZ(2, 0, 0, 2, 0, 0, 2, 0, 0);
+    intersectionPoint = ray.intersectionWithTriangle(triangle.points[0], triangle.points[1], triangle.points[2]);
+    ck.testUndefined(
+      intersectionPoint,
+      "expect no intersection when we have a degenerate triangle with three equal vertexes"
+    );
+
+    origin = Point3d.create(0, 0, 0);
+    direction = Vector3d.create(1, 1, 0);
+    ray = Ray3d.create(origin, direction);
```

Comment:
I need to redefine triangle here:

triangle = BarycentricTriangle.createXYZXYZXYZ(2, 0, 0, 10, 0, 0, 2, 10, 0);

I added it locally and the test passed.

I also added a test for ray intersects inside a triangle and it passed too. Will added them all after you reviewed this.